### PR TITLE
JsonSerializer - Fixed bug when handling custom IConvertible returning TypeCode.Empty

### PR DESCRIPTION
**Pull request by snakefoot**
Alternative of #4399 to fix invalid JSON format when JProperty is null

### CODE REVIEW

**Comment by snakefoot** [comment]
@vicentegnz No idea. It depends on when @304NotModified finds the optimal time-window.

File: src/NLog/Internal/StringHelpers.cs [hunk 1]
```diff
@@ -51,10 +51,7 @@ public static class StringHelpers
         internal static bool IsNullOrWhiteSpace(string value)
         {
 #if NET3_5
-
-            if (value == null) return true;
-            if (value.Length == 0) return true;
-            return String.IsNullOrEmpty(value.Trim());
+            return value?.Length > 0 ? string.IsNullOrEmpty(value.Trim()) : true;
```
Comment (304NotModified) [inline src/NLog/Internal/StringHelpers.cs:54]: Do you find this really easier to read?

Comment (snakefoot) [reply src/NLog/Internal/StringHelpers.cs:54]: Yes I do, and the compiler also think it is better. But I can guess from your leading question, that I should change it back?

Comment (304NotModified) [reply src/NLog/Internal/StringHelpers.cs:54]: Hoe is this better for the compiler? I'm pretty sure that an equal comparison is better than a larger than in term of performance

Comment (snakefoot) [reply src/NLog/Internal/StringHelpers.cs:54]: Less instructions are usually better for the compiler, but I will revert. Since this is not important.

**Review by 304NotModified — APPROVED** [review]

**Comment by 304NotModified** [comment]
> When will version 4.7.10 be released? will it come with this change, or is it postponed for later?

Will release it tomorrow :)

**Comment by snakefoot** [comment]
[NLog v4.7.10](https://www.nuget.org/packages/NLog/4.7.10) has been released.

**Pull request merged**